### ISSUE TITLE
M2.1.2 : Database / configuration seems lost after automated deployment.

### ISSUE THREAD

**Issue by kanduvisla** [magento/magento2#7169]
I'm not even sure how to report this issue, because it seems very random and the fix also works very random, but I'll give it a try:
### Fact 1:

I've got a deployment setup that creates a copy of the database, runs a upgrade/compile/static-content against it, and if everything passes it simply copies all the generated stuff from `pub/static` and `var/di` and `var/generation` to the production site and runs a `setup:upgrade --keep-generated` afterwards. This way I can do an automatic deployment whilst minimising downtime.
### Fact 2:

This method works well for 3 Magento 2 shops I have at the moment.
### However:

Yesterday I had a strange error where the homepage rendered only the message `There was no Home CMS page configured or found.`. Also, none of the navigation rendered, etc. etc. The shop was just blank. I could log in in the admin, and everything was configured just fine. I looked in the database and everything was just fine.

Needless to say, running `bin/magento cache:clean` solved the issue.

This night another deployment was run, and the same thing happened. Only this time, a simple `bin/magento cache:clean` didn't do anything. I also couldn't login in the admin. I tried various things, running `setup:upgrade --keep-generated` again, removing cache and page cache folders, nothing worked.

Then I tried `bin/magento cache:flush`. Now the site showed the message that there was an error with my request. The error log looked like this:

```
a:4:{i:0;s:146:"Warning: array_keys() expects parameter 1 to be array, null given in /data/web/build-master/vendor/magento/module-eav/Model/Config.php on line 511";i:1;s:5784:"#0 [internal function]: Magento\Framework\App\ErrorHandler->handler(2, 'array_keys() ex...', '/data/web/build...', 511, Array)
#1 /data/web/build-master/vendor/magento/module-eav/Model/Config.php(511): array_keys(NULL)
#2 /data/web/build-master/vendor/magento/module-eav/Model/Entity/AttributeLoader.php(90): Magento\Eav\Model\Config->getEntityAttributeCodes(Object(Magento\Eav\Model\Entity\Type), NULL)
#3 /data/web/build-master/vendor/magento/module-eav/Model/Entity/AttributeLoader.php(77): Magento\Eav\Model\Entity\AttributeLoader->checkAndInitAttributes(Object(Magento\Catalog\Model\ResourceModel\Category\Interceptor), NULL)
#4 /data/web/build-master/vendor/magento/module-eav/Model/Entity/AbstractEntity.php(510): Magento\Eav\Model\Entity\AttributeLoader->loadAllAttributes(Object(Magento\Catalog\Model\ResourceModel\Category\Interceptor), NULL)
#5 /data/web/build-master/vendor/magento/module-eav/Model/Entity/Collection/AbstractCollection.php(484): Magento\Eav\Model\Entity\AbstractEntity->loadAllAttributes()
#6 /data/web/build-master/vendor/magento/module-catalog/Model/AbstractModel.php(241): Magento\Eav\Model\Entity\Collection\AbstractCollection->addAttributeToSelect('*')
#7 /data/web/build-master/app/code/Vendor/Module/App/Router/NoRouteHandler.php(77): Magento\Catalog\Model\AbstractModel->loadByAttribute('url_key', 'home')
#8 /data/web/build-master/vendor/magento/framework/App/Router/DefaultRouter.php(46): Vendor\Module\App\Router\NoRouteHandler->process(Object(Magento\Framework\App\Request\Http))
#9 /data/web/build-master/vendor/magento/framework/App/FrontController.php(50): Magento\Framework\App\Router\DefaultRouter->match(Object(Magento\Framework\App\Request\Http))
#10 /data/web/build-master/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#11 /data/web/build-master/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#12 /data/web/build-master/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'requestPreproce...')
#13 /data/web/build-master/vendor/magento/module-store/App/FrontController/Plugin/RequestPreprocessor.php(94): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#14 /data/web/build-master/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\FrontController\Plugin\RequestPreprocessor->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#15 /data/web/build-master/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'install')
#16 /data/web/build-master/vendor/magento/framework/Module/Plugin/DbStatusValidator.php(69): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#17 /data/web/build-master/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Framework\Module\Plugin\DbStatusValidator->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#18 /data/web/build-master/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#19 /data/web/build-master/vendor/magento/module-page-cache/Model/App/FrontController/VarnishPlugin.php(55): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#20 /data/web/build-master/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\PageCache\Model\App\FrontController\VarnishPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#21 /data/web/build-master/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#22 /data/web/build-master/vendor/magento/module-page-cache/Model/App/FrontController/BuiltinPlugin.php(73): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#23 /data/web/build-master/vendor/magento/framework/Interception/Interceptor.php(142): Magento\PageCache\Model\App\FrontController\BuiltinPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#24 /data/web/build-master/var/generation/Magento/Framework/App/FrontController/Interceptor.php(26): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#25 /data/web/build-master/vendor/magento/framework/App/Http.php(135): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#26 /data/web/build-master/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#27 /data/web/build-master/pub/index.php(37): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#28 {main}";s:3:"url";s:10:"/home.html";s:11:"script_name";s:10:"/index.php";}
```

_Small notice: the `Vendor/Module`-part in the error report can be ignored, it's a WIP-module that ATM only extends it's parent._
### The strange part

This is what I did to resolve the issue:
- I edited `app/etc/env.php` and replaced the name of the database with that of a staging database.
- `bin/magento cache:clean` : no success
- I reverted `app/etc/env.php` back to it's original state. So it's now back to what it was.
- `bin/magento cache:clean` : problem solved
- What the heck?

So to summarize: I tried `bin/magento cache:clean` a couple of times, but it wasn't until I edited `env.php` and reverted it back to it's original state to make it solve the problem.

So... is this an issue? Has anyone else encountered this before? What might be the problem here?

**Comment by maderlock** [comment]
We have a similar deployment process, but have seen the white page issue at random points, not necessarily at deployment, so it may not be the only trigger.

**Comment by kanduvisla** [comment]
Had the same problem today. Very strange.

The site seemed to be unable to load the database. Or at least, parts of it. `bin/magento cache:flush` didn't solve it. But here's how I fixed it:

- I edited `env.php`.
- I added a linebreak at the end (just pressed enter).
- Save it.
- Flushed the cache.
- It works.

**Comment by shiftedreality** [comment]
Hi @kanduvisla 

I was unable to reproduce this issue.
Please provide more information about your environment-especially caching.

**Comment by vzabaznov** [comment]
Hi @kanduvisla , closed as we don't receive update more than 2 week, feel free to write here/reopen or to open new issue

**Comment by kanduvisla** [comment]
We still have this problem from time to time. Once again this morning in a 2.1.2 installation.

Anyone else encountered this issue?
